rbi/stdlib/csv.rbi

```diff
@@ -267,13 +267,20 @@ class CSV < Object
   # would read UTF-32BE data from the file but transcode it to UTF-8 before
   # [`CSV`](https://docs.ruby-lang.org/en/2.6.0/CSV.html) parses it.
   sig do
-    type_parameters(:U).params(
+    params(
         path: T.any(String, ::Sorbet::Private::Static::IOLike),
-        options: T::Hash[Symbol, T.type_parameter(:U)],
-        blk: T.proc.params(arg0: T::Array[T.nilable(String)]).void,
+        options: T::Hash[Symbol, BasicObject],
+        blk: T.proc.params(arg0: T.any(T::Array[T.nilable(BasicObject)], CSV::Row)).void,
```

Comment:
Would you mind changing this here and in `parse` with `T.any(T::Array[T.untyped], CSV::Row)` instead?

Consider the following example:

```ruby
total = 0
CSV.parse(file_content, {converters: %i[numeric]}) do |row|
  col1, _ = row
  total += col1
end
```

With your change, there'd be a type error on `total += col1`, because Sorbet has no way of knowing the type of `col1`. Given that the type is specified as `BasicObject`, you can't even call `to_i` on it, so you are left with no other option but to `T.cast`ing it 😢

What's more, there are no guarantees as to what a `transformer` might return, as callers are free to send a lambda that can return whatever type they'd like.